snorkel/learning/pytorch/noise_aware_model.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from builtins import *
+
+import os
+from time import time
+
+import numpy as np
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+from snorkel.learning.classifier import Classifier
+from snorkel.learning.utils import reshape_marginals, LabelBalancer
+
+
+def cross_entropy_loss(input, target):
+    total_loss = torch.tensor(0.0)
+    for i in range(input.size(1)):
+        cls_idx = torch.full((input.size(0),), i, dtype=torch.long)
+        loss = F.cross_entropy(input, cls_idx, reduce=False)
+        total_loss += target[:, i].dot(loss)
+    return total_loss / input.shape[0]
+
+
+class TorchNoiseAwareModel(Classifier, nn.Module):
+    """
+    Generic NoiseAwareModel class for PyTorch models.
+    
+    :param n_threads: Parallelism to use; single-threaded if None
+    :param seed: Top level seed which is passed into both numpy operations
```

Comment:
This doesn't seem to actually be handled anywhere?  Also this is a fun place of trickiness... need to make sure to seed any other random modules (`random, numpy.random`) as well as potentially give options to turn off CuDNN for determinism, e.g. see https://github.com/HazyResearch/metal_nips/blob/666eea29ecbc9f816e3e9ae7690200409610ff09/metal/metal_model.py#L205